src/protocol/keyexpr.c

```diff
@@ -103,7 +102,10 @@ bool _z_keyexpr_equals(const _z_keyexpr_t *left, const _z_keyexpr_t *right) {
     if (_z_keyexpr_mapping_id(left) != _z_keyexpr_mapping_id(right)) {
         return false;
     }
-    return _z_string_equals(&left->_suffix, &right->_suffix);
+    if (_z_keyexpr_has_suffix(left) || _z_keyexpr_has_suffix(right)) {
```

Comment:
Will better to add a case for _z_keyexpr_has_suffix(left) && _z_keyexpr_has_suffix(right), because as far as I see, _z_string_equals will have UB in case of keyexpr with NULL suffix

Reply:
Then to take all possible cases it should be something like:
```C
if (_z_keyexpr_has_suffix(left) != _z_keyexpr_has_suffix(right)) {
    return false;
}
if (_z_keyexpr_has_suffix(left) && _z_keyexpr_has_suffix(right)) {
    return _z_string_equals(&left->_suffix, &right->_suffix);
}
```